ocaml/xapi/test_xapi_vbd_helpers.ml

```diff
@@ -0,0 +1,70 @@
+(*
+ * Copyright (C) 2006-2012 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open OUnit
+open Test_common
+
+(* Helpers for testing Xapi_vbd_helpers.valid_operations *)
+
+let setup_test ~__context ~vdi_fun =
+  let _sm_ref = make_sm ~__context () in
+  let sr_ref = make_sr ~__context () in
+  let (_: API.ref_PBD) = make_pbd ~__context ~sR:sr_ref () in
+  let vm_ref = make_vm ~__context () in
+  let vdi_ref = vdi_fun sr_ref in
+  let vbd_ref = make_vbd ~vDI:vdi_ref ~vM:vm_ref ~__context () in
+  let vbd_record = Db.VBD.get_record_internal ~__context ~self:vbd_ref in
+  vbd_ref, vbd_record
+
+let my_cmp a b = match a,b with
+  | Some(code1, _), Some(code2, _) -> code1 = code2
+  | None, None -> true
+  | _	-> false
+
+let string_of_api_exn_opt = function
+  | None -> "None"
+  | Some (code, args) ->
+    Printf.sprintf "Some (%s, [%s])" code (String.concat "; " args)
+
+let run_assert_equal_with_vdi ~__context ?(cmp = my_cmp) ?(expensive_sharing_checks=false) ~vdi_fun op exc =
```

Comment:
I don't know about the details but I would expect that we can afford to run expensive tests in unit tests.